lib/active_fedora/file/fedora_attributes.rb

```diff
@@ -0,0 +1,39 @@
+module ActiveFedora::File::FedoraAttributes
+
+  attr_writer :mime_type
+
+  def mime_type
+    @mime_type ||= fetch_mime_type unless new_record?
+    @mime_type || default_mime_type
+  end
+
+  def original_name
+    @original_name ||= fetch_original_name_from_headers
+  end
+
+  def original_name= name
+    @original_name = name
+  end
+
+  def digest
+    response = metadata.ldp_source.graph.query(predicate: ActiveFedora::RDF::Fcrepo4.digest)
+    response.map(&:object)
+  end
+
+  protected
+
+    def default_mime_type
+      'text/plain'
+    end
+
+    def fetch_mime_type
```

Comment:
I think these two methods can be private.  Also, can we change the name of one of them so they both either use `_from_headers` or both don't?